container.go

```diff
@@ -63,24 +63,23 @@ func (c *Container) Provide(t interface{}) error {
 	defer c.Unlock()
 
 	ctype := reflect.TypeOf(t)
-
 	switch ctype.Kind() {
 	case reflect.Func:
-		switch ctype.NumOut() {
-		case 2:
-			if ctype.Out(1) != _typeOfError {
-				return errReturnErrKind
-			}
-			fallthrough
-		case 1:
+		count := ctype.NumOut()
+		if count == 0 {
+			return errReturnCount
+		} else if count == 1 {
 			objType := ctype.Out(0)
 			if objType.Kind() != reflect.Ptr && objType.Kind() != reflect.Interface {
 				return errReturnKind
 			}
-		default:
-			return errReturnCount
+		} else {
+			// last variable must be error
+			if ctype.Out(count-1) != _typeOfError {
```

Comment:
Oh, I thought the point of multi args return is to make the last variable strictly an `error` type. (At least that was the behavior with 2 parameter constructor). Will update.